modyn/supervisor/internal/triggers/datadrifttrigger.py

```diff
@@ -180,21 +159,42 @@ def inform(
             # Reset for next detection
             self._sample_left_until_detection = self.config.detection_interval_data_points
 
-            # The first ever detection will always trigger
-            if not self._triggered_once or len(self.warmup_intervals) < (self.config.warmup_intervals or 0):
-                # If we've never triggered before, always trigger
-                self._triggered_once = True
-                triggered = True
+            if (not self._triggered_once) or (
+                not self.warmup_completed and len(self.warmup_intervals) < (self.config.warmup_intervals or 0)
+            ):
+                # Warmup trigger, evaluate warmup policy while storing the reference window
+                # for later calibration with the drift policy.
+
+                # delegate to the warmup policy
+                delegated_trigger_results = (
+                    next(
+                        self.warmup_trigger.inform([(idx, time, 0) for (idx, time) in next_detection_interval]),
+                        None,
+                    )
+                    if self.warmup_trigger
+                    else None
+                )
+                triggered = (
+                    delegated_trigger_results is not None
+                    if self._triggered_once
+                    else True  # first candidate always triggers
+                )
+
                 drift_results: dict[str, MetricResult] = {}
                 if len(self.warmup_intervals) < (self.config.warmup_intervals or 0):
-                    self.warmup_intervals.append(list(self._windows.reference))
+                    # for the first detection the reference window is empty, therefore adding the current window
+                    self.warmup_intervals.append(
+                        list(self._windows.reference if self._triggered_once else self._windows.current)
+                    )
+
+                self._triggered_once = True
```

Comment:
I am wondering about the case where

1) we dont have warmup (=> triggered once is checked)
2) we specify a warmup trigger

what if that warmup trigger does not trigger?

then we set triggered once to true, despite potentially not triggering. is that intended?

Reply:
if the warmup policies doesn't trigger in the first interval, that doesn't matter, for the first interval we force a trigger

```python
triggered = (
    delegated_trigger_results is not None
    if self._triggered_once
    else True  # first candidate always triggers
)
```